test/unit_tests/braket/default_simulator/test_state_vector_simulator.py

```diff
@@ -1363,3 +1369,11 @@ def test_rotation_parameter_expressions(operation, state_vector):
     result = simulator.run(OpenQASMProgram(source=qasm), shots=0)
     assert result.resultTypes[0].type == StateVector()
     assert np.allclose(result.resultTypes[0].value, np.array(state_vector))
+
+
+def test_discontiguous_qubits(discontiguous_jaqcd):
+    prg = JaqcdProgram.parse_raw(discontiguous_jaqcd)
+    result = StateVectorSimulator().run(prg, qubit_count=2, shots=1)
+
+    assert result.measuredQubits == [0, 1]
+    assert result.measurements == [["1", "1"]]
```

Comment:
Please add tests for OpenQASM as well

Reply:
Would OpenQASM spec support discontiguous qubits? I tried with
```python
qasm = """
bit[2] b;
qubit[2] q;
h q[2];
cnot q[2], q[9];
b = measure q;
 """
```

and got error
```python
qubit = 2

    def validate_qubit_in_range(qubit: int):
        if qubit >= len(self[name]):
>           raise IndexError(
                f"qubit register index `{qubit}` out of range for qubit register of length {len(self[name])} `{name}`."
            )
E           IndexError: qubit register index `2` out of range for qubit register of length 2 `q`.

src/braket/default_simulator/openqasm/program_context.py:120: IndexError
```

Reply:
In OpenQASM, non-contigious qubits would manifest as only a proper subset of the declared qubit register being used:
```
qasm = """
bit[2] b;
qubit[3] q;
h q[2];
cnot q[2], q[9];
b = measure q;
"""
```

Reply:
I tried this test using the example above but it fails to compile the OpenQASM definition. Would I be missing something?

```
def test_discontiguous_qubits_openqasm():
        qasm = """
        bit[2] b;
        qubit[3] q;
        h q[2];
        cnot q[2], q[9];
        b = measure q;
        """
        simulator = StateVectorSimulator()
>       result = simulator.run(OpenQASMProgram(source=qasm), shots=0)
```

error
```
qubit = 9

    def validate_qubit_in_range(qubit: int):
        if qubit >= len(self[name]):
>           raise IndexError(
                f"qubit register index `{qubit}` out of range for qubit register of length {len(self[name])} `{name}`."
            )
E           IndexError: qubit register index `9` out of range for qubit register of length 3 `q`.

src/braket/default_simulator/openqasm/program_context.py:120: IndexError
```

Reply:
Sorry, typo; I meant
```
qasm = """
bit[2] b;
qubit[13] q;
h q[2];
cnot q[2], q[9];
b = measure q;
"""
```
should work